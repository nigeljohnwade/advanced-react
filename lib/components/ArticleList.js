import React from 'react';
import Article from './Article';

const ArticleList = (props) => {
    return (
        <div>
            <h1>Articles</h1>
            {Object.values(props.articles).map((article) =>
                <Article
                    key={article.id}
                    article={article} />
            )}
        </div>
    );
};

export default ArticleList;