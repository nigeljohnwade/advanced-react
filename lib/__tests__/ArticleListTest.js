import React from 'react';
import ArticleList from '../components/ArticleList';
import { shallow } from 'enzyme';

describe('ArticleList', () => {
    const testProps = {
        articles: {
            a: { id: 'a', date: 'date', title: 'date', body: 'date' },
            b: { id: 'b', date: 'date', title: 'date', body: 'date' }
        },

    };
    it('renders correctly', () => {
        const wrapper = shallow(
            <ArticleList
                {...testProps}
            />
        );
        expect(wrapper.find('ArticleContainer').length).toBe(2);
        expect(wrapper).toMatchSnapshot();
    });

});