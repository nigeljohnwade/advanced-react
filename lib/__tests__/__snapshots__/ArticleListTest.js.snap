// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <h1>
    Articles
  </h1>
  <div
    style={
      Object {
        "marginBottom": "1em",
      }
    }
  >
    <h2
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    />
    <p
      style={
        Object {
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
    </p>
    <p
      style={
        Object {
          "fontSize": "0.75em",
        }
      }
    >
      <a
        href={undefined}
      >
         
      </a>
    </p>
    <p
      style={undefined}
    />
  </div>
  <div
    style={
      Object {
        "marginBottom": "1em",
      }
    }
  >
    <h2
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    />
    <p
      style={
        Object {
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
    </p>
    <p
      style={
        Object {
          "fontSize": "0.75em",
        }
      }
    >
      <a
        href={undefined}
      >
         
      </a>
    </p>
    <p
      style={undefined}
    />
  </div>
</div>
`;
